tools/restatectl/src/commands/metadata/migrate.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use anyhow::Context;
+use bytes::Bytes;
+use clap::Parser;
+use clap_stdin::FileOrStdin;
+use cling::{Collect, Run};
+
+use restate_cli_util::_comfy_table::Table;
+use restate_cli_util::c_println;
+use restate_cli_util::ui::console::StyledTable;
+use restate_core::protobuf::cluster_ctrl_svc::{MigrateMetadataRequest, new_cluster_ctrl_client};
+use restate_types::config::MetadataClientOptions;
+use restate_types::nodes_config::Role;
+
+use crate::commands::metadata::MetadataCommonOpts;
+use crate::connection::ConnectionInfo;
+
+#[derive(Run, Parser, Collect, Clone, Debug)]
+#[clap()]
+#[cling(run = "migrate")]
+pub struct MigrateOpts {
+    #[clap(flatten)]
+    metadata: MetadataCommonOpts,
+
+    /// The TOML configuration of the target metadata store
+    #[arg(long)]
+    toml: Option<FileOrStdin>,
+
+    /// The Json configuration of the target metadata store
+    #[arg(long, conflicts_with = "toml")]
+    json: Option<FileOrStdin>,
```

Comment:
Should we stick with one way to provide the configuration? Otherwise, we give users different ways to do the same thing and we need to explain how to specify `MetadataClientOptions` in json and in toml. Maybe sticking with `toml` is the easiest since users already use it when configuring Restate via the `config.toml`?